### ISSUE TITLE
How to use this method for generating novel research ideas and new hypothesis for the science domain of Physics?

### ISSUE THREAD

**Issue by jbdatascience** [NoviScl/AI-Researcher#13]
Having skimmed your very interesting research paper []("Can LLMs Generate Novel Research Ideas?
A Large-Scale Human Study with 100+ NLP Researchers (https://arxiv.org/pdf/2409.04109v1)"), I was impressed by what is already possible at this moment in time!
As far as I can see now, the research was concentrated on generating novel research ideas in the domain of Natural Language Processing (NLP).

So immediately the next question came with me:
**How to use this method for generating novel research ideas and new hypothesis for the science domain of Physics?**

I cannot oversee what parts of your method need to be adapted to be able to do that.
Although I can imagine at least one thing in that regard: perhaps the LLM model(s) used for generating novel research ideas be better finetuned already on existing information from that domain! 
But also I can imagine that something should be changed in the prompts that are used? Perhaps that is not necessary?

So the main question I have is:
How to use this method for generating novel research ideas and new hypothesis for the science domain of Physics? What parts of the code and/or prompts in [](https://github.com/NoviScl/AI-Researcher) have to be changed in order to be able to do that? 

Perhaps other changes are necessary?

**Comment by jimcost** [comment]
I suspect the effort will be on a decent size. Take a look at the `prompts` and `src`, for example, because there is "Natural Language Process Researcher" like descriptions in there. I suspect migrating it over will take some diligence. A place to start would be to `grep` things like "langauge" and see what files come up.

**Comment by NoviScl** [comment]
Yeah we customized some of our prompts specifically for generating NLP research ideas, so those prompts need to be changed for your specific research domain. 

Also, you probably need to change the few-shot demo examples as well - I hand-crafted the current demo examples from existing NLP papers, and you probably want to find a few example physics papers as demo examples for your case.
